### ISSUE TITLE
flit-core: pyproject.toml missing dependencies on toml and tomli

### ISSUE THREAD

**Issue by adamjstewart** [pypa/flit#482]
@takluyver I'm updating Spack's flit and flit-core packages and I noticed that flit-core's pyproject.toml does not list any dependencies. However, tomli (and possibly toml?) seem to be required to use flit-core. Is that correct? Should these be explicitly listed in the pyproject.toml? I realize that there's a cyclic dependency between flit-core and tomli, is this the reason this dependency isn't listed?

**Comment by takluyver** [comment]
You're right, `tomli` is required (it replaced `toml` from Flit version 3.4).

There's actually no metadata at all in flit_core's `pyproject.toml`, precisely because we'd need a TOML parser to read it. The metadata that would normally be there is embedded in the code instead:

https://github.com/pypa/flit/blob/ec6e6b7c8994d5e79d0fd951d9e4c26a7c5d9977/flit_core/flit_core/build_thyself.py#L19-L36

The cyclic dependency is a bit of a pain. My current best idea for how to deal with it is described on [the bootstrapping page](https://flit.readthedocs.io/en/latest/bootstrap.html). But there's a (long) ongoing discussion about this, e.g. #462, #481, #483.

**Comment by adamjstewart** [comment]
Thanks, that explains it. I think it still might be useful to include in the `pyproject.toml` for humans to read, but I'll leave that up to you. For what it's worth, in Spack our bootstrapping strategy is to download a copy of the flit-core source code when building tomli but not installing flit-core, and having a dependency on tomli in the flit-core package.

**Comment by pradyunsg** [comment]
The other option is putting a comment in the pyproject.toml file, stating where to find the data + a link to the above comment. :)

Do you want to file a PR?
